rust/cubesqlplanner/cubesqlplanner/src/logical_plan/join.rs

```diff
@@ -0,0 +1,74 @@
+use super::pretty_print::*;
+use super::Cube;
+use super::SimpleQuery;
+use crate::planner::sql_evaluator::{MemberSymbol, SqlCall};
+use std::rc::Rc;
+
+#[derive(Clone)]
+pub struct CubeJoinItem {
+    pub cube: Rc<Cube>,
+    pub on_sql: Rc<SqlCall>,
+}
+
+impl PrettyPrint for CubeJoinItem {
+    fn pretty_print(&self, result: &mut PrettyPrintResult, state: &PrettyPrintState) {
+        result.println(&format!("CubeJoinItem for cube: {}", self.cube.name), state);
+    }
+}
+
+#[derive(Clone)]
+pub struct SubqueryDimensionJoinItem {
+    pub subquery: Rc<SimpleQuery>,
+    pub dimension: Rc<MemberSymbol>,
+    pub primary_keys_dimensions: Vec<Rc<MemberSymbol>>,
+}
+
+impl PrettyPrint for SubqueryDimensionJoinItem {
+    fn pretty_print(&self, result: &mut PrettyPrintResult, state: &PrettyPrintState) {
+        result.println(
+            &format!(
+                "SubqueryDimensionJoinItem for dimension `{}`: ",
+                self.dimension.full_name()
+            ),
+            state,
+        );
+        result.println("subquery:", state);
+        result.println("primary_keys_dimensions:", state);
+        let state = state.new_level();
+        for dim in self.primary_keys_dimensions.iter() {
+            result.println(&format!("- {}", dim.full_name()), &state);
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum LogicalJoinItem {
```

Comment:
Are any new variants expected or planned here?

Reply:
Yep, PreAggregationSource for example.

Reply:
Yep, pre aggregation for example